src/nvidia/src/kernel/gpu/fifo/kernel_channel.c

```diff
@@ -3280,10 +3280,11 @@ kchannelGetChildIterator
 (
     KernelChannel *pKernelChannel,
     NvU32 classID,
-    NvU32 engineID,
+    RM_ENGINE_TYPE type,
     KernelChannelChildIterator *pIter
 )
 {
+    NvU32 engineID = (NvU32)type;
```

Comment:
No need to have an NvU32 variable, we can use the RM_ENGINE_TYPE argument directly

Reply:
In that case I get this:
```
make -C src/nvidia
make[1]: Entering directory '/home/tuommaki/projects/nanovms.gpu-nvidia/src/nvidia'
 [ nvidia            ]  CC           src/kernel/gpu/fifo/kernel_channel.c
src/kernel/gpu/fifo/kernel_channel.c:3279:1: error: conflicting types for ‘kchannelGetChildIterator’ due to enum/integer mismatch; have ‘void(KernelChannel *, NvU32,  NvU32,  KernelChannelChildIterator *)’ {aka ‘void(KernelChannel *, unsigned int,  unsigned int,  KernelChannelChildIterator *)’} [-Werror=enum-int-mismatch]
 3279 | kchannelGetChildIterator
      | ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from inc/kernel/gpu/fifo/kernel_channel.h:2,
                 from src/kernel/gpu/fifo/kernel_channel.c:28:
generated/g_kernel_channel_nvoc.h:1519:6: note: previous declaration of ‘kchannelGetChildIterator’ with type ‘void(struct KernelChannel *, NvU32,  RM_ENGINE_TYPE,  KernelChannelChildIterator *)’ {aka ‘void(struct KernelChannel *, unsigned int,  RM_ENGINE_TYPE,  KernelChannelChildIterator *)’}
 1519 | void kchannelGetChildIterator(struct KernelChannel *pKernelChannel,
      |      ^~~~~~~~~~~~~~~~~~~~~~~~
cc1: note: unrecognized command-line option ‘-Wno-unknown-warning-option’ may have been intended to silence earlier diagnostics
cc1: all warnings being treated as errors
make[1]: *** [Makefile:224: _out/Nanos_x86_64/kernel_channel.o] Error 1
make[1]: Leaving directory '/home/tuommaki/projects/nanovms.gpu-nvidia/src/nvidia'
make: *** [Makefile:35: src/nvidia/_out/Nanos_x86_64/nv-kernel.o] Error 2
```

See: https://github.com/nanovms/gpu-nvidia/blob/main/src/nvidia/generated/g_kernel_channel_nvoc.h#L1521

Reply:
I didn't mean keeping the NvU32 engineID parameter in the function definition, my comment was just about the NvU32 engineID local variable you added, which is not needed: you can use directly the RM_ENGINE_TYPE function argument.